### PR TITLE
Added identification ID of the provided option type by its sku in the specified store.

### PR DESCRIPTION
**Pull request by magelogger**
Description:
1. Added optionTypeSkus private property to the class app/code/Magento/CatalogImportExport/Model/Import/Product/Option.php

2. Added $optionSku parameter to the getExistingOptionTypeId function.

3. Added identification ID of the provided option type by its sku in the specified store.

Fixed Issues
1. magento/magento2#23601: Magento2 Impossible to update custom option value based on custom option SKU

Manual testing scenarios:
1.Create product with Customizable Options
2. Check option_type_id in table catalog_product_option_type_value
3. Export this product
4. Edit in csv file custom options title 
5. Save scv file and import it
6. Check option_type_id in table catalog_product_option_type_value

Expected result:
Value option_type_id is not changed when row is updated

### CODE REVIEW

**Comment by sidolov** [comment]
Hi @magelogger , please, sign CLA, otherwise, we can't process your pull request

File: app/code/Magento/CatalogImportExport/Model/Import/Product/Option.php [hunk 1]
```diff
@@ -1625,14 +1636,15 @@ private function restoreOriginalOptionTypeIds(array &$typeValues, array &$typePr
...
+     * Identify ID of the provided option type by its title or sku in the specified store.
      *
      * @param int $optionId
      * @param int $storeId
      * @param string $optionTypeTitle
+     * @param string $optionSku
      * @return int|null
      */
-    private function getExistingOptionTypeId($optionId, $storeId, $optionTypeTitle)
+    private function getExistingOptionTypeId($optionId, $storeId, $optionTypeTitle, $optionSku)
```
Comment (aleron75) [inline app/code/Magento/CatalogImportExport/Model/Import/Product/Option.php:1656]: When applicable, I would assign types to function parameters as long as to the return value.
In this case, the method is private and we wouldn't break any backward compatibility. What do you think?

**Comment by VladimirZaets** [comment]
@magento run all tests

**Comment by VladimirZaets** [comment]
Hi @magelogger  , please, sign CLA, otherwise, we can't process your pull request

**Review by VladimirZaets — CHANGES_REQUESTED** [review]
Hi @magelogger  , please, sign CLA, otherwise, we can't process your pull request

**Comment by engcom-Charlie** [comment]
Hi @magelogger, I'm closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for your collaboration.

**Pull request closed without merging**